### PR TITLE
Closes the terminal when websocket disconnects

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8223/

**Comment by jaycee** [comment]
QA:

This is a pain.

You need to deploy the jujushell charm, with updated jujushell and termserver resources in a model on jujucharms.com or staging.jujucharms.com.

Connect to the model with the gui using guiproxy, setting `gisf` to `true` and `flags: {jujushell: true}`.

Then expose the jujushell charm to enable the terminal.

When you type `exit` and hit enter, or hit ctrl+d, the terminal should close. When you open a new terminal it should be a new session.

Once in the terminal, hit ctrl

File: jujugui/static/gui/src/app/components/terminal/terminal.js [hunk 1]
```diff
@@ -76,6 +76,9 @@ class Terminal extends React.Component {
         });
         return;
       }
+      if (resp['0'] === 'disconnect') {
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/terminal/terminal.js]: Maybe add a comment explaining that this is sent by terminado when the process it runs exits.

**Comment by jaycee** [comment]
:shipit:

File: jujugui/static/gui/src/app/components/terminal/terminal.js (hunk 1 above)
Comment (jaycee) [reply jujugui/static/gui/src/app/components/terminal/terminal.js]: Done. Thanks!

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8227/

**Pull request merged**